functional-test/src/test/groovy/org/rundeck/tests/functional/api/job/JobActivityHistorySpec.groovy

```diff
@@ -0,0 +1,54 @@
+package org.rundeck.tests.functional.api.job
+
+import org.rundeck.util.annotations.APITest
+import org.rundeck.util.container.BaseContainer
+
+@APITest
+class JobActivityHistorySpec extends BaseContainer {
+    def setupSpec() {
+        startEnvironment()
+
+    }
+
+    def "Retrieve job Activity history from Job List Page "() {
+        given: "A Job exists in the project with execution history"
+        def jobId = "3957f735-5269-4c85-bc57-f123598886f6"
+
+        when: "The activity history for the job is requested"
+
+        def response = doGet("/job/${jobId}/executions")
+        def expectedKeys = [
+                'id', 'status', 'date-started', 'date-ended', 'user', 'project'
+        ]
+        then: "The response should contain  a list of execution details for the Job"
+        verifyAll {
+            response.successful
+            response.code() == 200
+            def json = jsonValue(response.body(), Map)
+            json.executions.every { execution ->
+                def executionMap = execution as Map
+                println "Execution Details: ${executionMap}"
+                expectedKeys.every { key ->
+                    executionMap.containsKey(key)
+
+                }
+            }
```

Comment:
Keep in mind the AC of your tickets.
https://pagerduty.atlassian.net/browse/RUN-2644?atlOrigin=eyJpIjoiN2YwM2VkYzcxMWM5NGVhNDg4NWUwNTUzNzk0NjBiMGIiLCJwIjoiaiJ9

You're writing (UI) "selenium" test that very "job history"

Reply:
> Keep in mind the AC of your tickets. https://pagerduty.atlassian.net/browse/RUN-2644?atlOrigin=eyJpIjoiN2YwM2VkYzcxMWM5NGVhNDg4NWUwNTUzNzk0NjBiMGIiLCJwIjoiaiJ9
> 
> You're writing (UI) "selenium" test that very "job history"

Thanks, I understand that I need to follow selenium based test structure for this. I am reviewing the existing selenium tests for guidance and update my code accordingly. It's great to see the browser interaction during the tests.